ci_framework/roles/cifmw_cephadm/tasks/dashboard/dashboard.yml

```diff
@@ -0,0 +1,117 @@
+      changed_when: false
+
+    - name: import dashboard certificate key
+      ansible.builtin.command: "{{ cifmw_cephadm_ceph_cli }} config-key set mgr/dashboard/key -i {{ cifmw_cephadm_dashboard_key }}"
+      changed_when: false
+  when: cifmw_cephadm_dashboard_crt | length > 0 and
+        cifmw_cephadm_dashboard_key | length > 0
+
+- include_tasks: configure_dashboard_backends.yml
+  with_items: "{{ groups['edpm'] }}"
+  vars:
+    dashboard_backend: '{{ item }}'
+
+- name: Restart the Ceph dashboard
+  become: true
+  block:
+    - name: disable mgr dashboard module (restart)
+      ansible.builtin.command: "{{ cifmw_cephadm_ceph_cli }} mgr module disable dashboard"
+    - name: enable mgr dashboard module (restart)
+      ansible.builtin.command: "{{ cifmw_cephadm_ceph_cli }} mgr module enable dashboard"
+
+- name: Configure Monitoring Stack
+  become: true
+  vars:
+    grafana_host:  "{{ groups['edpm'][0] }}"
+  block:
+    - name: get Grafana instance address
+      ansible.builtin.set_fact:
+        grafana_server_addr: "{{ hostvars[grafana_host][all_addresses] | ansible.utils.ipaddr(cifmw_cephadm_monitoring_network) | first }}"
+    - name: set grafana api user
+      ansible.builtin.command: "{{ cifmw_cephadm_ceph_cli }} dashboard set-grafana-api-username {{ cifmw_cephadm_grafana_admin_user }}"
+    - name: set grafana api password
+      ansible.builtin.command: "{{ cifmw_cephadm_ceph_cli }} dashboard set-grafana-api-password -i -"
+      args:
+        stdin: "{{ cifmw_cephadm_grafana_admin_password }}"
+        stdin_add_newline: no
+    - name: disable ssl verification for grafana
+      ansible.builtin.command: "{{ cifmw_cephadm_ceph_cli }} dashboard set-grafana-api-ssl-verify False"
+      changed_when: false
+      when:
+        - cifmw_cephadm_dashboard_crt | length > 0 and
+          cifmw_cephadm_dashboard_key | length > 0
+
+    - name: set alertmanager host
+      ansible.builtin.command: |
+        {{ cifmw_cephadm_ceph_cli }} dashboard set-alertmanager-api-host http://{{ grafana_server_addr }}:9093
+    - name: set prometheus host
+      ansible.builtin.command: |
+        {{ cifmw_cephadm_ceph_cli }} dashboard set-prometheus-api-host \
+        http://{{ grafana_server_addr }}:{{ cifmw_cephadm_prometheus_port }}
+    - name: config grafana api url
+      ansible.builtin.command: |
+        {{ cifmw_cephadm_ceph_cli }} dashboard set-grafana-api-url \
+        http://{{ grafana_server_addr }}:{{ cifmw_cephadm_grafana_port }}
+
+- name: Restart the Ceph dashboard
+  become: true
+  block:
+    - name: disable mgr dashboard module (restart)
+      ansible.builtin.command: "{{ cifmw_cephadm_ceph_cli }} mgr module disable dashboard"
+    - name: enable mgr dashboard module (restart)
+      ansible.builtin.command: "{{ cifmw_cephadm_ceph_cli }} mgr module enable dashboard"
+
+- name: 20 second pause for dashboard services to be up and running.
+  ansible.builtin.pause:
+    seconds: 20
```

Comment:
should this be a generic configurable parameter?

Reply:
In my testing, i observed that we need a mandatory little sleep here as the previous task will disable and enable the dashboard.
but it cannot be a configurable parameter as it is should be set the best value (min time needed for dashboard url to be ready)
with 10 seconds, it failed, 
with 20 , it failed once in many runs
with 30 , it passed every time, so '30' should be ideal value.

Reply:
ack, but I'm wondering if, instead of having a sleep time here, we can remove this `validating` task and add it to a "post.yaml" action where we can collect a set of validations.
In that case you know for sure that at least 20 seconds passed, and you don't need the sleep task anymore.